### PR TITLE
feat(autocomplete): added autocomplete & fixed some other issues

### PR DESCRIPTION
**Pull request by sudoku-lord**
Finally putting up my PR! There was a testing issue which was deceptively difficult to figure out, but after a day of poking around and a lot of debug prints, I found it: I had to include `content-type` within the header construction in `setup.js`.

Other small changes include adding a `LOB_API_LIVE_TOKEN` in order to successfully hit the autocomplete endpoint and get a response. In fact, as it turns out, address verification also requires a live key in order to verify an address, so I changed it in the `us_verifications` test too––but if we decide that we don't need that specific functionality within the scope of these tests, I can change it back. I also corrected one test in `zip_lookups_test.js`.

### CODE REVIEW

File: tests/zip_lookups_test.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ test("use an incorrectly formatted zip code", async function (t) {
       )
     );
 
-  await t.doesNotReject(Promise.resolve(response));
+  await t.rejects(Promise.reject(response));
```
Comment (hilary) [inline tests/zip_lookups_test.js:42]: Nice!

**Review by hilary — CHANGES_REQUESTED** [review]
Almost there...

File: resources/us_autocompletions/models/us_autocompletions.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  address_prefix:
+    type: string
+    description: >
+      The prefix of an address's primary line. This only supports street names & primary numbers,
+      and only accepts alphanumeric characters.
+
+  city:
+    allOf:
+      - $ref: "../../../shared/attributes/address/city.yml"
+      - description: Matches partial abbreviations.
```
[the remark this answers is not in the document]
Comment (sudoku-lord) [reply resources/us_autocompletions/models/us_autocompletions.yml:16]: Oh no. That's a copypasta gah.

File: resources/us_autocompletions/models/us_autocompletions.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  state:
+    allOf:
+      - $ref: "../../../shared/attributes/address/state.yml"
+      - description: >
+          An optional ZIP Code input used to filter suggestions. Case insensitive, and does not match
+          partial abbreviations.
```
[the remark this answers is not in the document]
Comment (sudoku-lord) [reply resources/us_autocompletions/models/us_autocompletions.yml:23]: yup oops

File: resources/us_autocompletions/models/us_autocompletions.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+  zip_code:
+    allOf:
+      - $ref: "../../../shared/attributes/address/zip_code.yml"
+      - description: >
+          An optional ZIP Code input used to filter suggestions. Matches partial abbreviations.
```
[the remark this answers is not in the document]
Comment (sudoku-lord) [reply resources/us_autocompletions/models/us_autocompletions.yml:29]: a copypasta!

File: resources/us_autocompletions/models/us_autocompletions.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    type: string
+    description: >
+      An optional ZIP Code input used to filter suggestions. Matches partial entries.
+
+  geo_ip_sort:
+    type: string
+    description: >
+      If true, sort suggestions by proximity to the IP set in the `X-Forwarded-For` header.
+
+  object:
```
Comment (hilary) [inline resources/us_autocompletions/models/us_autocompletions.yml:34]: I'm sorry, I'm not very with it today. I actually saw this issue first time around but then lost track of it. Vaccinated brain is not happy brain.

The `object` property isn't a payload parameter. I'm not entirely sure why we provide it, but that's another story. I looked, and we made the same error in `us_verifications`. Please remove `object` from those payload schemas as well.

File: resources/us_autocompletions/responses/us_autocompletions.yml [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+        suggestions:
+          type: array
+          description: >
+            An array of objects representing suggested addresses.
+          minItems: 0
+          maxItems: 10
+          items:
+            $ref: "../models/suggestions.yml"
+          readOnly: true
```
Comment (hilary) [inline resources/us_autocompletions/responses/us_autocompletions.yml:27]: we're not using `readOnly` as it is going away in `3.1.0`

**Review by hilary — CHANGES_REQUESTED** [review]
I found a couple more things, sorry.

1. In `Lob-API-public.yml`, you need to add `us_autocompletions` to the appropriate group in `x-tagGroups` for the resource to show.
2. Do any other resources reference `us_auto_id`? Seems unlikely as we only have international verifications left to go. :smile: We're only putting ids that are referenced by multiple resources in `shared/attributes/model_ids/`, ids used only by the resource itself go in `resources/<resource_name>/attributes/<id_name>.yml`. I realized that the CONTRIBUTING doc is out of date and am working on an update.
3. a couple of other small items I commented on inline.

File: resources/us_autocompletions/responses/us_autocompletions.yml (hunk 6 above)
Comment (sudoku-lord) [reply resources/us_autocompletions/responses/us_autocompletions.yml:27]: Ah yup, that's copypasta

**Review by hilary — APPROVED** [review]
:shipit:

**Pull request merged**